omnibus/files/private-chef-cookbooks/private-chef/libraries/preflight_checks.rb

```diff
@@ -126,7 +126,12 @@ def initialize(node)
   def run!
     begin
       BootstrapPreflightValidator.new(node).run!
-      PostgresqlPreflightValidator.new(node).run!
+      # When Chef Backend is configured, this is too early to verify
+      # postgresql accessibility since we need to configure HAProxy
+      # first
+      if ! PrivateChef['use_chef_backend']
```

Comment:
With this flag, we can at some point  get rid of the hacky "derive whether we really care about bootstrap" logic that's used to allow a front-end to do bootstrapping.